lib/storage/metadata/mongoclient/ListRecordStream.js

```diff
@@ -65,9 +66,7 @@ class ListRecordStream extends stream.Readable {
         this._lastConsumedID = itemObj.h.toString();
 
         const decode = tags => (
-            JSON.parse(JSON.stringify(tags).
-                       replace(/\uFF0E/g, '.').
-                       replace(/\uFF04/g, '$'))
+            MongoUtils.unescape(tags)
         );
```

Comment:
If decode is just shorthand for unescape you do not need the extra function.

```javascript
const decode = MongoUtils.unescape
```

Reply:
Can even get shorter:
``
const unescape = require('./utils').unescape;
``